pkg/kernel/networkservice/inject/server.go

```diff
@@ -30,119 +30,114 @@ import (
 	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
 	"github.com/networkservicemesh/sdk/pkg/tools/log"
 
-	"github.com/networkservicemesh/sdk-kernel/pkg/kernel/utils"
+	"github.com/networkservicemesh/sdk-kernel/pkg/kernel/nsswitch"
 )
 
-type injectServer struct {
-}
+type injectServer struct{}
 
 // NewServer - returns a new networkservice.NetworkServiceServer that moves given network interface into the Client's
 // pod network namespace on Request and back to Forwarder's network namespace on Close
 func NewServer() networkservice.NetworkServiceServer {
 	return &injectServer{}
 }
 
-func (a *injectServer) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*networkservice.Connection, error) {
+func (s *injectServer) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*networkservice.Connection, error) {
+	logEntry := log.Entry(ctx).WithField("injectServer", "Request")
+
 	connID := request.GetConnection().GetId()
 	mech := kernel.ToMechanism(request.GetConnection().GetMechanism())
 
-	nsSwitch, err := utils.NewNSSwitch()
+	nsSwitch, clientNetNSHandle, err := initNetNSSwitchAndHandle(mech.GetNetNSURL())
```

Comment:
It internally do lock, it is hard to read, could you please improve this moment. Thanks.

Reply:
Actually there is only a need to wrap GetNS, SetNS with lock, so I have moved all locking stuff to the NSSwitch internals.